server/internal/core/application/service.go

```diff
@@ -639,22 +643,23 @@ func (s *covenantlessService) SubmitOffchainTx(
 		signedCheckpointTxs[rebuiltCheckpointTx.UnsignedTx.TxID()] = signedCheckpointTx
 	}
 
-	expiration, roundTxid := findFirstRoundToExpire(spentVtxos)
+	expiration, rootCommitmentTxid := findFirstRoundToExpire(spentVtxos)
 	if expiration == 0 {
 		return nil, "", "", fmt.Errorf("invalid expiry")
 	}
 
-	commitmentTxids := []string{roundTxid}
-	delete(indexedCommitmentTxids, roundTxid)
-	for txid := range indexedCommitmentTxids {
-		commitmentTxids = append(commitmentTxids, txid)
-	}
-
-	change, err := offchainTx.Accept(signedRedeemTx, signedCheckpointTxs, commitmentTxids, expiration)
+	change, err := offchainTx.Accept(
+		signedRedeemTx,
+		signedCheckpointTxs,
+		commitmentTxsByCheckpointTxid,
+		rootCommitmentTxid,
+		expiration,
+	)
 	if err != nil {
 		return nil, "", "", fmt.Errorf("failed to accept offchain tx: %s", err)
 	}
 	changes = append(changes, change)
+	s.offchainTxs.add(*domain.NewOffchainTxFromEvents(changes))
```

Comment:
not needed to replay the events

```suggestion
	s.offchainTxs.add(offchainTx)
```